e2e/key/keyprivate/keyprivate.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package keyprivate
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sylabs/singularity/e2e/internal/keyexec"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+
+var keyPath string
+var defaultKeyFile string
+
+func testPrivateKeyNewPair(t *testing.T) {
+	tests := []struct {
+		name    string
+		user    string
+		email   string
+		note    string
+		psk1    string
+		psk2    string
+		push    bool
+		succeed bool
+	}{
+		{
+			name:    "newpair",
+			user:    "genbye2etests\n",
+			email:   "westley@sylabs.io\n",
+			note:    "test key generated by e2e tests\n",
+			psk1:    "e2etests\n",
+			push:    false,
+			succeed: true,
+		},
+		// TODO: add a couple more tests
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			t.Run("remove_private_keyring_before_newpair", test.WithoutPrivilege(func(t *testing.T) { keyexec.RemoveKeyring(t) }))
+			c, b, err := keyexec.KeyNewPair(t, tt.user, tt.email, tt.note, tt.psk1, tt.psk2, tt.push)
+			if tt.succeed {
+				if err != nil {
+					t.Log("Command that failed: ", c)
+					t.Log(string(b))
+					t.Fatalf("unexpected failure: %v", err)
+				}
+				keyexec.QuickTestExportImportKey(t)
```

Comment:
I'm going to guess that QuickTestExportImportKey _depends_ on being called exactly here.

What I mean is that since this function receives only t as parameter, it's tightly coupled with the key ring that is being created or modified here.

Reply:
I also included another parameter for this function, that will tell `QuickTestExportImportKey` what key to test.